include/llvm/Support/WinFunctions.h

```diff
@@ -0,0 +1,81 @@
+//===-- WinFunctions.h - Windows Functions for other platforms --*- C++ -*-===//
+//
+//                     The LLVM Compiler Infrastructure
+//
+// This file is distributed under the University of Illinois Open Source
+// License. See LICENSE.TXT for details.
+//
+//===----------------------------------------------------------------------===//
+//
+// This file defines Windows-specific functions used in the codebase for
+// non-Windows platforms.
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef LLVM_SUPPORT_WINFUNCTIONS_H
+#define LLVM_SUPPORT_WINFUNCTIONS_H
+
+#ifndef _WIN32
+
+static HRESULT StringCchPrintfA(char *dst, size_t dstSize, const char *format,
```

Comment:
Just as an FYI. I have a warning-killing change that puts the bodies of these into a source file because every time they are included in full in a header and not used, they produce a warning. If you wanted, you could do the same as a "while I'm at it" change.

Reply:
We should definitely make this change.